test/unit/org/apache/cassandra/security/DefaultSslContextFactoryTest.java

```diff
@@ -53,15 +53,23 @@ private void addKeystoreOptions(Map<String,Object> config)
         config.put("keystore_password", "cassandra");
     }
 
+    private void addOutboundKeystoreOptions(Map<String, Object> config)
+    {
+        config.put("outbound_keystore", "test/conf/cassandra_ssl_test.keystore");
+        config.put("outbound_keystore_password", "cassandra");
+    }
+
     @Test
     public void getSslContextOpenSSL() throws IOException
     {
-        EncryptionOptions options = new EncryptionOptions().withTrustStore("test/conf/cassandra_ssl_test.truststore")
-                                                           .withTrustStorePassword("cassandra")
-                                                           .withKeyStore("test/conf/cassandra_ssl_test.keystore")
-                                                           .withKeyStorePassword("cassandra")
-                                                           .withRequireClientAuth(false)
-                                                           .withCipherSuites("TLS_RSA_WITH_AES_128_CBC_SHA");
+        EncryptionOptions.ServerEncryptionOptions options = new EncryptionOptions.ServerEncryptionOptions().withTrustStore("test/conf/cassandra_ssl_test.truststore")
+                                                                                                           .withTrustStorePassword("cassandra")
+                                                                                                           .withKeyStore("test/conf/cassandra_ssl_test.keystore")
+                                                                                                           .withKeyStorePassword("cassandra")
+                                                                                                           .withOutboundKeystore("test/conf/cassandra_ssl_test.keystore")
+                                                                                                           .withOutboundKeystorePassword("cassandra")
+                                                                                                           .withRequireClientAuth(false)
+                                                                                                           .withCipherSuites("TLS_RSA_WITH_AES_128_CBC_SHA");
         SslContext sslContext = SSLFactory.getOrCreateSslContext(options, true, ISslContextFactory.SocketType.CLIENT);
         Assert.assertNotNull(sslContext);
```

Comment:
How do you know if you got the right certificate here? From the regular keystore or the outbound keystore?

Reply:
From the created SslContext we won't know anything about the keystore or the certificate. One thing we can do is to use a `mockito.spy` object to check if `buildKeyManagerFactory` or `buildOutboundKeyManagerFactory` has been called.

Reply:
@jonmeredith  added test cases `testServerSocketShouldUseKeystore` & `testClientSocketShouldUseOutboundKeystore` to verify that the SSLContext has the expected certificates